libcontainer/cgroups/fs/cpuset.go

```diff
@@ -95,6 +95,10 @@ func (s *CpusetGroup) ensureParent(current, root string) error {
 	if filepath.Clean(parent) == root {
 		return nil
 	}
+	// Avoid infinite recursion.
+	if parent == current {
+		return nil
+	}
```

Comment:
Can you be more specific, like how can we reproduce this with docker? I still don't know how could `filepath.Dir(path) == path` and it's not returned by "filepath.Clean(parent) == root".

And I think it'll help to add more comments for such code.